File: app/src/main/java/com/kksionek/photosyncer/SyncApplication.java
package com.kksionek.photosyncer;

import androidx.multidex.MultiDexApplication;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class SyncApplication extends MultiDexApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        Realm.init(this);
        RealmConfiguration config = new RealmConfiguration.Builder()
                .name("photosync.realm")
                .deleteRealmIfMigrationNeeded()
                .build();
        Realm.setDefaultConfiguration(config);
    }
}

File: app/src/main/java/com/kksionek/photosyncer/SyncApplication.kt
package com.kksionek.photosyncer

import android.app.Application
import dagger.hilt.android.HiltAndroidApp

import io.realm.Realm
import io.realm.RealmConfiguration

@HiltAndroidApp
class SyncApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        Realm.init(this)
        val config = RealmConfiguration.Builder()
            .name("photosync.realm")
            .deleteRealmIfMigrationNeeded()
            .build()
        Realm.setDefaultConfiguration(config)
    }
}
